### ISSUE TITLE
[BE] 10-10CG - Improvements: Update Form1010cg::Service to use new OAuth 2.0 Authorization Flow from MuleSoftClient in Dev

### ISSUE THREAD

**Issue by coope93** [department-of-veterans-affairs/va.gov-team#86425]
As a first step for implementing OAuth 2.0 Authentication for the 10-10CG form, we need to update the existing MuleSoftClient to use the new authentication flow. 

- Should be behind flipper toggle
- Ensure we update all VCR cassettes as the api payload will be different with this change. 
- Update any documentation around 10-10 CG and MuleSoft 
- More than likely this will follow a client credential grant flow (confirmed this is the case) as it is service to service authentication.
- Helpful info for OAuth 2.0 Client Credential flow with Okta is [here](https://developer.okta.com/docs/guides/implement-oauth-for-okta-serviceapp/main/)
- SI (DEV/QA) or DTC-I team (Stage/Prod) will create the application through AnyPoint Exchange for the API which is requested access by the consumer (va.gov)
- DTC-I team will provide us the auth server URL, Client Id, and Client Secret in an encrypted email. (These will need added to AWS Parameter Store with a platform support ticket)

## Acceptance Criteria
- Development is complete and manually tested successfully, coordinating with Mulesoft for Dev credentials.  Additional testing in Staging and potentially code changes due to testing are captured in a separate ticket.

**Comment by coope93** [comment]
Currently blocked on this while I wait for the MuleSoft team to get me staging credentials.

**Comment by coope93** [comment]
Confirmed today that test submissions from dev were showing up in CARMASIT dev environment. 🥳
